src/mlpack/bindings/python/mlpack/matrix_utils.py

```diff
@@ -55,7 +55,10 @@ def to_matrix(x, dtype=np.double, copy=False):
   else:
     if isinstance(x, pd.core.series.Series):
       x = pd.DataFrame(x)
-    return np.array(x, copy=True, dtype=dtype, order='C'), True
+    if copy: # Copy the matrix if required.
+      return np.array(x, copy=True, dtype=dtype, order='C'), True
+    else:
+      return np.array(x, dtype=dtype), False
```

Comment:
What happens if we pass a Pandas Dataframe where the order is `F` instead of `C`?  Try this code:

```
import pandas as pd
import numpy as np
d = pd.read_csv("some_data.csv")
x = np.array(d)
x.flags
```

You can see that `F_CONTIGUOUS` is true, but we need `C_CONTIGUOUS` to be true when we pass directly to mlpack.  So I think we need a way to handle this.

In addition I think this still copies the Pandas dataframe, since when I look at `x.flags` I see

```
  C_CONTIGUOUS : False                                                                                                                                          
  F_CONTIGUOUS : True                                                                                                                                           
  OWNDATA : True                                                                                                                                                
  WRITEABLE : True                                                                                                                                              
  ALIGNED : True                                                                                                                                                
  WRITEBACKIFCOPY : False                                                                                                                                       
  UPDATEIFCOPY : False   
```

and `OWNDATA : True` implies that the matrix was copied.

Also, we should probably add a test case to `src/mlpack/bindings/python/tests/test_python_binding.py`.  Take a look at the test `testNumpyMatrix(self)`, and we can adapt this to use a Pandas dataframe.  That would be a good way to ensure this works.